ui/components/multichain/edit-networks-modal/edit-networks-modal.js

```diff
@@ -12,13 +17,75 @@ import {
   Checkbox,
   Text,
   Box,
+  ModalFooter,
+  ButtonPrimary,
+  ButtonPrimarySize,
 } from '../../component-library';
 import { NetworkListItem } from '..';
+import {
+  grantPermittedChains,
+  removePermittedChain,
+} from '../../../store/actions';
 
-export const EditNetworksModal = ({ onClose }) => {
+export const EditNetworksModal = ({ onClose, onClick }) => {
   const t = useI18nContext();
+  const dispatch = useDispatch();
   const nonTestNetworks = useSelector(getNonTestNetworks);
   const testNetworks = useSelector(getTestNetworks);
+  const chains = useSelector(getPermittedChainsByOrigin);
+  const permittedChains = Object.values(chains);
+  const activeTabOrigin = useSelector(getOriginOfCurrentTab);
+  const flattenedPermittedChains = permittedChains.flat();
+  const [selectedChains, setSelectedChains] = useState(
+    flattenedPermittedChains,
+  );
+
+  const handleAccountClick = (chainId) => {
+    const index = selectedChains.indexOf(chainId);
+    let newSelectedChains = [];
+
+    if (index === -1) {
+      // If chainId is not already selected, add it to the selectedChains array
+      newSelectedChains = [...selectedChains, chainId];
+    } else {
+      // If chainId is already selected, remove it from the selectedChains array
+      newSelectedChains = selectedChains.filter((_item, idx) => idx !== index);
+    }
+    setSelectedChains(newSelectedChains);
+  };
+  const allAreSelected = () => {
+    return nonTestNetworks.length === selectedChains.length;
+  };
+  let checked = false;
+  let isIndeterminate = false;
+  if (allAreSelected()) {
+    checked = true;
+    isIndeterminate = false;
+  } else if (selectedChains.length > 0 && !allAreSelected()) {
+    checked = false;
+    isIndeterminate = true;
+  }
```

Comment:
Can be simplified like this:

```
const checked = allAreSelected();
const isIndeterminate = selectedChains.length > 0 && !checked;
```

Reply:
Updated the logic and we need it to allAreSelected, with checked it doesn't seem to work

Reply:
https://github.com/MetaMask/metamask-extension/pull/26635/commits/d065a0cb67631ccf420c3d27c1896ff9552cbc3e